lib/src/server/server.dart

```diff
@@ -174,6 +174,11 @@ class Server extends ConnectionServer {
       socket.setOption(SocketOption.tcpNoDelay, true);
       final connection = ServerTransportConnection.viaSocket(socket,
           settings: http2ServerSettings);
+      if (security != null && !security.validateClient(socket)) {
```

Comment:
I guess we should write a simple test for this as well given how easy we have accidentally lost it (I did not notice it in my review, sorry).